### PR TITLE
Command to create Notification

### PR DESCRIPTION
**Pull request by aquibbaig**
Initial PR which is aimed at fixing #388

### CODE REVIEW

File: site/src/Command/CreateNotificationCommand.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        $this->userManager = $userManager;
+    }
+
+    /**
+     * Configuration for the Command
+     */
+    protected function configure()
+    {
+        $this->setDescription('Adds a Notification to the database')
+            ->setHelp('This command allows you to send a web notification to Rabbitmq.');
```
Comment (imphil) [inline site/src/Command/CreateNotificationCommand.php]: This tool is neither limited to rabbitmq nor to web notifications. How about "Send a notification to a user" as description?

File: site/src/Command/CreateNotificationCommand.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        $this->addArgument('subject', InputArgument::REQUIRED, 'Add a Notification subject...');
+        $this->addArgument('message', InputArgument::REQUIRED, 'Add a Notification message...');
+        $this->addArgument('type', InputArgument::REQUIRED, 'Specify the Notification type...');
```
Comment (imphil) [inline site/src/Command/CreateNotificationCommand.php]: - Please remove all those "...". 
- "Notification" isn't capitalized.
- You need to add a "username" argument.

File: site/src/Command/CreateNotificationCommand.php [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+     * Publish notifications for test users
+     *
+     * @param InputInterface  $input
+     * @param OutputInterface $output
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $users = $this->userManager->findUsers();
+
+        foreach ($users as $user) {
```
Comment (imphil) [inline site/src/Command/CreateNotificationCommand.php]: Don't send a notification to all users, add a "username" argument, use the UserManager to actually get the `User` for a given username, and send the notification to this one only.

File: site/src/Command/CreateNotificationCommand.php [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+     *
+     * @param ProducerInterface    $notificationProducer
+     * @param Notification         $notification
+     * @param UserManagerInterface $userManager
+     */
+    public function __construct(ProducerInterface $notificationProducer, Notification $notification, UserManagerInterface $userManager)
+    {
+        parent::__construct();
+        $this->producer = $notificationProducer;
+        $this->notification = $notification;
```
Comment (imphil) [inline site/src/Command/CreateNotificationCommand.php]: Why is `notification` an object member variable? It should be a local variable in the `execute()` method.

File: site/src/Command/CreateNotificationCommand.php [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    {
+        $this->setDescription('Adds a Notification to the database')
+            ->setHelp('This command allows you to send a web notification to Rabbitmq.');
+        $this->addArgument('subject', InputArgument::REQUIRED, 'Add a Notification subject...');
+        $this->addArgument('message', InputArgument::REQUIRED, 'Add a Notification message...');
+        $this->addArgument('type', InputArgument::REQUIRED, 'Specify the Notification type...');
+    }
+
+    /**
+     * Publish notifications for test users
```
Comment (imphil) [inline site/src/Command/CreateNotificationCommand.php]: Adjust this comment as well when adding the `username` input argument.

File: site/src/Command/CreateNotificationCommand.php [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+use Doctrine\Common\Collections\ArrayCollection;
+use FOS\UserBundle\Model\UserManagerInterface;
+use OldSound\RabbitMqBundle\RabbitMq\ProducerInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Input\InputArgument;
+
+/**
+ * Creates a Notification and publishes it to RabbitMQ
```
Comment (imphil) [inline site/src/Command/CreateNotificationCommand.php]: Create a notification for a given user.

File: site/src/Command/CreateNotificationCommand.php [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+ */
+class CreateNotificationCommand extends Command
+{
+    /**
+     * @var string
+     */
+    protected static $defaultName = 'create:notification';
```
Comment (imphil) [inline site/src/Command/CreateNotificationCommand.php]: Should be librecores:send-notification

File: site/src/Command/CreateNotificationCommand.php [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    private $userManager;
+
+    /**
+     * CreateNotificationCommand constructor.
+     *
+     * @param ProducerInterface    $notificationProducer
+     * @param Notification         $notification
+     * @param UserManagerInterface $userManager
+     */
+    public function __construct(ProducerInterface $notificationProducer, Notification $notification, UserManagerInterface $userManager)
```
Comment (imphil) [inline site/src/Command/CreateNotificationCommand.php]: Why do you inject the Notification data class here?

**Comment by aquibbaig** [comment]
@imphil Yep. Thanks for the review. I just made some changes

**Comment by aquibbaig** [comment]
@imphil `user` -> `username`

**Comment by imphil** [comment]
Thanks!

**Pull request merged**